qutebrowser/mainwindow/mainwindow.py

```diff
@@ -492,10 +492,13 @@ def _connect_signals(self):
     @pyqtSlot(bool)
     def _on_fullscreen_requested(self, on):
         if on:
-            self.state_before_fullscreen = self.windowState()
-            self.showFullScreen()
+            self.window_state_before_fullscreen = self.windowState()
+            self.config_state_before_fullscreen = config.val.content.desktop_fullscreen
```

Comment:
You'll also need to initialize this in `__init__` correctly. I'm also not sure if it makes sense to name those so similarly, as they are two completely different things.

Reply:
I've gone ahead and removed that second variable and check because I'm now thinking it's not really unexpected behaviour.